pkg/upgraders/osd/upgrader.go

```diff
@@ -589,19 +589,34 @@ func performClusterHealthCheck(c client.Client, metricsClient metrics.Metrics, c
 		return false, fmt.Errorf("unable to query critical alerts: %s", err)
 	}
 
-	if len(alerts.Data.Result) > 0 {
-		logger.Info("There are critical alerts exists, cannot upgrade now")
-		return false, fmt.Errorf("there are %d critical alerts", len(alerts.Data.Result))
+	alertCount := len(alerts.Data.Result)
+
+	if alertCount > 0 {
+		alert := []string{}
+		uniqueAlerts := make(map[string]bool)
+
+		for i := range alerts.Data.Result {
+			a := alerts.Data.Result[i].Metric["alertname"]
```

Comment:
why you use the index to get the result here, but not use something like below?

```
for i, r := range alerts.Data.Result {
          a := r.Metric["alertname"]
}
```

Reply:
Thanks Meng Bo!! Yes, agreed that's a better approach. Change commit is pushed.